File: 프로그래머스/3/389481. 봉인된 주문/봉인된 주문.py
def solution(n, bans):
    nums = list(map(ban2num, bans))
    
    want = n
    for num in sorted(nums):
        if num <= want:
            want += 1
    
    print(want)
    return num2ban(want)


def ban2num(ban):
    ret = 0
    for b in ban:
        ret *= 26
        ret += ord(b) - ord('a') + 1
    return ret


def num2ban(num):
    ret = ''
    while num:
        num -= 1
        ret += chr(ord('a') + num % 26)
        num //= 26
    return ret[::-1]

File: 프로그래머스/3/389481. 봉인된 주문/봉인된 주문.java
import java.util.*;

class Solution {
    public String solution(long n, String[] bans) {
        long[] nums = Arrays.stream(bans)
            .mapToLong(this::banToNum)
            .sorted()
            .toArray();
        
        long want = n;
        for (long num: nums) {
            if (num <= want)
                want++;
        }
        
        return numToBan(want);
    }
    
    public long banToNum(String ban) {
        long ret = 0L;
        for (char c: ban.toCharArray()) {
            ret *= 26;
            ret += c - 'a' + 1;
        }
        
        return ret;
    }
    
    public String numToBan(Long num) {
        StringBuilder ret = new StringBuilder();
        while (num > 0) {
            num -= 1;
            ret.append((char)(num % 26 + 'a'));
            num /= 26;
        }
        return ret.reverse().toString();
    }
}